compose-builder/add-security.yml

```diff
@@ -106,18 +106,33 @@ services:
       - edgex-network
     ports:
       - "127.0.0.1:8200:8200"
-    cap_add:
-      - "IPC_LOCK"
+    deploy:
+      resources:
+        limits:
+          memory: "1024m"
+    memswap_limit: "1024m"
     tmpfs:
       - /vault/config
-    entrypoint: ["/edgex-init/vault_wait_install.sh"]
+    entrypoint: [ "/edgex-init/vault_wait_install.sh" ]
     env_file:
       - common-sec-stage-gate.env
     command: server
     environment:
       VAULT_ADDR: http://edgex-vault:8200
       VAULT_CONFIG_DIR: /vault/config
       VAULT_UI: "true"
+      VAULT_LOCAL_CONFIG: >
+        listener "tcp" { 
+            address = "edgex-vault:8200" 
+            tls_disable = "1" 
+            cluster_address = "edgex-vault:8201" 
+        }
+        backend "file" {
+            path = "/vault/file"
+        }
+        default_lease_ttl = "168h" 
+        max_lease_ttl = "720h"
+        disable_mlock = true
```

Comment:
Does this change also depend on docker rootless?

Reply:
No. Vault will run fine in both rootless and rootful docker environments. I have tested this